lib/idp_common_pkg/idp_common/summarization/service.py

```diff
@@ -405,6 +405,13 @@ def process_document(
             # Update document status to completed if not already failed
             if document.status != Status.FAILED:
                 document.status = Status.COMPLETED
+            try:
+                # Perserving Document Summary for Frontend Across DeCoupled Architecture
```

Comment:
Please clarify the purpose of this new logic.. the comment doesn't explain it clearly to me.